tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -57,3 +57,87 @@ def test_repeated_iteration_constants():
               [15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0]]
     config = {"operation": "constants", "name": "key", "values": values}
     repeated_iteration(config)
+
+
+def test_attack_point_iterator_balanced_generator():
+    config = {
+        "operation": "balanced_generator",
+        "name": "key",
+        "kwargs": {
```

Comment:
as mentioned in the DD, i'm not a fan of this `kwargs`. First that python-related so `params` would be better suited. And second, it's an unnecessary level of indirection where the parameters could move one level up, at the same level as `operation` and `name`.